internal/handler/handler.go

```diff
@@ -61,3 +76,67 @@ func (svc htmlPdfService) Ping(w http.ResponseWriter, r *http.Request) {
 	response.ToJson(w, resp.Status, resp.Message, resp.Data)
 	return
 }
+func (svc htmlPdfService) Upload(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseMultipartForm(10240) //File size to come from config
+	if err != nil {
+		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrFileSizeExceeded), nil)
+		log.Error(err.Error())
+		return
+	}
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrFileParseFail), nil)
+		log.Error(err.Error())
+		return
+	}
+	defer file.Close()
+	resp := svc.logic.Upload(file)
+	response.ToJson(w, resp.Status, resp.Message, resp.Data)
+}
+func (svc htmlPdfService) ConvertToPdf(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	//we take id as a parameter from url path
+	id, ok := vars["id"]
+	if !ok {
+		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrIdNotfound), nil)
```

Comment:
Instead of giving the error as id not found, a more appropriate error would be "id needed". Create a new error code and use it when looking for id in the URL path.

Reply:
done